usr/local/share/bastille/etcupdate.sh

```diff
@@ -0,0 +1,254 @@
+#!/bin/sh
+#
+# Copyright (c) 2018-2023, Rodrigo Nascimento Hernandez <rodrigomdev@gmail.com>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+. /usr/local/share/bastille/common.sh
+. /usr/local/etc/bastille/bastille.conf
+
+usage() {
+        # Update /etc folder while keeping user changes
+        error_notify "Usage: bastille etcupdate [option(s)(optional)] [jailname] [oldrelease] [newrelease]"
+
+        cat << EOF
+        Options:
+
+        -D | --dryrun   -- Do a dry run. Output actions to stdout but without making changes.
+        -Q | --quiet    -- Do not output actions to stdout
+
+EOF
+        exit 1
+}
+
+
+executeconditional() {
+    if [ $DRY_RUN -eq "0" ]; then
+        eval "$@"
+    fi
+}
+
+C1_C6_conditions() {
+    filelistjail=$(find "${jail_etc}" -mindepth 1 -type f)
+    for jailfile in ${filelistjail}
```

Comment:
The pattern of `for X in $(find...)` is risky if filenames contain whitespace or control chars, but the "usual" mechanisms to sanely deal with arbitrary filenames are typically `bash`-specific (e.g. `find ... -print0 | while IFS= read -r -d''; do ... done`).

Generally, in POSIX shells, the only supported way is to `find -exec sh -c '...' +` or `find -print0 | xargs -0 sh -c '...'`, which would introduce a subshell and affect availability of your variables.

Perhaps, this may be better written as...

```
search_dir () {
  [ -n "$1" ] && return 1
  [ -d "$1" ] || return 1

  for jailfile in "${1}/*"; do
    if [ -d "${jailfile}" ]; then
      search_dir "${jailfile}"
    else
      process_jail_file "${jailfile}"
    fi
  done
}

search_dir "${jail_etc}"
```